src/main/java/com/linecorp/armeria/client/routing/ServerGroupRegistry.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.routing;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A memory server group register center.
+ */
+public class ServerGroupRegistry {
+    private static final Map<String, NodeSelector> serverGroups = new ConcurrentHashMap<>();
+
+    private ServerGroupRegistry() {
+    }
+
+    /**
+     * Register the server group.if the groupName has already registered,throw
+     * {@link ServerGroupException}.
+     */
+    public static boolean register(String groupName, ServerGroup serverGroup,
+                                   NodeSelectStrategy nodeSelectStrategy) {
+        requireNonNull(groupName, "groupName");
+        requireNonNull(serverGroup, "serverGroup");
+        requireNonNull(nodeSelectStrategy, "nodeSelectStrategy");
+
+        if (serverGroups.containsKey(groupName)) {
+            throw new ServerGroupException("A ServerGroup with the same name exists: " + groupName);
+        }
+
+        NodeSelector existSelector = serverGroups.putIfAbsent(groupName,
```

Comment:
Ugh, it should be:

    if (serverGroups.putIfAbsent(...) != null) { throw new ServerGroupException(...); }

No need to return a boolean value.